### DISCUSSION TITLE
How to revalidate a route from outside the route?

### DISCUSSION THREAD

**Discussion by hilja** [remix-run/remix#6080] [Q&A]
I've got a message queue where I processes some long lasting tasks. I can ask the queue which tasks are running and show a spinner in the front-end for the given items. Super handy.

When I send the task into the queue, my UI updates like it should, but since the task runs in the background Remix doesn't know it ends, and my spinners are spinning forever.

I've got a `completed` event in my message queue: `myQueue.on('completed', completedHandler)`, so the most optimal way for me would be to send a some sort of signal from the `completedHandler` to Remix asking it to revalidate the page (reload).

Could this be done somehow?

I've looked into `useRevalidator` but it seems like it only reacts to user input and other front-end events.

**Comment by hilja** [comment]
Of course it could be done with polling a resource route that lists all the queued items. Maybe start the polling when when user initiates the task and stop when there are no results. Polling is a bit extra though.

**Comment by hilja** [comment]
The polling I suggested is a terrible idea. If you'd want to do it [Sergio's post has a good example](https://sergiodxa.com/articles/automatic-revalidation-in-remix#revalidate-on-interval).

Would it be possible to POST to a route and have it trigger a `redirect('.')` in the action? If I want to do it with fetch from outside the routes I'd need to authenticate first.

**Reply by hilja** [reply]
Okay apparently you can't redirect to `.`, and also Remix just prints out the redirect command and does nothing when called from outside.

```
{
  connection: 'keep-alive',
  date: 'Sun, 16 Apr 2023 20:29:06 GMT',
  'keep-alive': 'timeout=5',
  'x-remix-redirect': '/redirected-path',
  'x-remix-status': '303'
}
```

**Reply by vaynevayne** [reply]
hello, what means `called from outside.` ?  has some code?

**Reply by hilja** [reply]
Outside the route file basically. A POST request from `app/utils/queue/consumer.ts`.

**Reply by juniorforlife** [reply]
@hilja have you been able to solve the issue?
My problem is I have a LanguageSwitcher in a Navbar. When I switch the language using remix-i18next, I want to reload the loader of any page that the user is currently at to switch the content to selected language. 
I tried revalidator but it doesn't seem to work at component level
